client/src/pages/Feed.js

```diff
@@ -319,11 +177,91 @@ const Feed = (props) => {
     });
     return ref.current;
   }
-  const { history, isAuthenticated, user, searchKeywords } = props;
+  const { history, isAuthenticated, user } = props;
 
   const dispatchAction = (type, key, value) =>
     feedDispatch({ type, key, value });
 
+  const getStateFromQuery = () => {
+    const query = qs.parse(history.location.search);
+
+    // search category (Tab)
+    query.s_category = SEARCH_OPTIONS[query.s_category]?.id || null;
+    changeHelpType(query.s_category);
+
+    // location
+    if (query.location) {
+      query.location = JSON.parse(atob(query.location));
+      dispatchAction(SET_VALUE, "location", query.location);
+    } else dispatchAction(SET_VALUE, "location", "");
+
+    // filters / help type (objective)
+    if (query.filters || query.objective) {
+      let selectedFilters = {};
+      if (query.filters) {
+        if (!query.s_category) {
+          query.filters = JSON.parse(atob(query.filters));
+          selectedFilters = query.filters;
+        } else delete query.filters;
+      }
+      if (query.objective) {
+        selectedFilters["lookingFor"] = [
+          filters[3].options[query.objective - 1]?.value,
+        ];
+      } else {
+        delete selectedFilters["lookingFor"];
+      }
+      optionsDispatch({
+        type: SET_OPTIONS,
+        payload: { option: selectedFilters },
+      });
+    } else {
+      optionsDispatch({ type: REMOVE_ALL_OPTIONS, payload: {} });
+    }
+    // will trigger => refetchPosts() =(if needed)> loadPosts()
+    setQueryParams(query);
+  };
+
+  const setQueryFromState = () => {
+    const newQuery = {};
+    const oldFiltersLength = [
+      ...(queryParams.filters?.type || []),
```

Comment:
why to spread? if you need to calc the length (array1.type || []).length + (array2.type || []).length